app/views/media/_media_form.html.erb

```diff
@@ -0,0 +1,16 @@
+<%= @error if @error %>
+<%= form_for @my_medium, method: @post_method, url: @post_path  do |f| %>
+
+	<%= f.hidden_field :kind, value: params[:kind] %>
```

Comment:
This should use `hidden_field_tag` instead of `f.hidden_field` because when we use `f.hidden_field` the resulting param is named `medium[kind]` and when we generate this form again (e.g. because there is a validation error when saving the new medium) we no longer have `param[:kind]`.

To reproduce this issue:

1. Click the link to create a new medium, then click "Submit Art" without filling in any fields.
1. You'll be shown the page again with the error message indicating you need to fill out all fields.
1. Fill out all fields and click "Submit Art" again.
1. Observe that you receive the error message again. This is because the re-rendered form no longer has `params[:kind]` so the hidden field is empty.

An alternative way to resolve this issue would be to change the controller and take out this hidden field entirely. See my next comment on the controller code for more details.